lib/plugin/swagger.js

```diff
@@ -18,76 +20,112 @@ const Policy = Joi.object({
+  variables: Joi.object().description('List of fixed values for variables').label('Variables')
 }).label('PolicyRef')
+const PolicyRefs = Joi.array().items(PolicyRef).description('Policy Refs').label('PolicyRefs')
 
 const UserRef = Joi.object({
   id: Joi.string().description('User ID'),
   name: Joi.string().description('User name')
 }).label('UserRef')
+const UserRefs = Joi.array().items(UserRef).description('User refs').label('UserRefs')
+
+const TeamRef = Joi.object({
+  id: Joi.string().description('Team ID'),
+  name: Joi.string().description('Team name')
+}).label('TeamRef')
+const TeamRefs = Joi.array().items(TeamRef).description('Team refs').label('TeamRefs')
 
 const Team = Joi.object({
   id: Joi.string().description('Team ID'),
   name: Joi.string().description('Team name'),
   description: Joi.string().description('Team description'),
   path: Joi.string(),
-  users: Joi.array().items(UserRef).description('Team users'),
-  policies: Joi.array().items(PolicyRef),
+  metadata: MetaData,
+  users: UserRefs,
+  policies: PolicyRefs,
   organizationId: Joi.string().description('Organization ID to which the team belongs to'),
   usersCount: Joi.number().description('Number of team users. Sub team users not counted.')
 }).label('Team')
 
-const TeamRef = Joi.object({
-  id: Joi.string().description('Team ID'),
-  name: Joi.string().description('Team name')
-}).label('TeamRef')
+const Teams = Joi.array().items(Team).description('items').label('Teams')
 
 const User = Joi.object({
   id: Joi.string().description('User ID'),
   name: Joi.string().description('User name'),
   organizationId: Joi.string().description('Organization ID to which the user belongs to'),
-  teams: Joi.array().items(TeamRef),
-  policies: Joi.array().items(PolicyRef)
+  metadata: MetaData,
+  teams: TeamRefs,
+  policies: PolicyRefs
 }).label('User')
+const Users = Joi.array().items(User).description('items').label('Users')
 
 const Organization = Joi.object({
   id: Joi.string().description('Organization ID'),
   name: Joi.string().description('Organization name'),
   description: Joi.string().description('Organization description'),
-  policies: Joi.array().items(PolicyRef)
+  metadata: MetaData,
+  policies: PolicyRefs
 }).label('Organization')
+const Organizations = Joi.array().items(Organization).description('items').label('Organizations')
 
 const OrganizationAndUser = Joi.object({
   organization: Organization,
   user: UserRef
 }).label('OrganizationAndUser')
 
-const List = (data) => {
+const POLICIES = 1
+const TEAMS = 2
+const TEAMREFS = 3
+const USERS = 4
+const ORGANIZATIONS = 5
+
+function getDataArray (listType) {
+  // todo: based on object return fixed type for swagger
+  if (listType === POLICIES) return Policies
+  if (listType === TEAMS) return Teams
+  if (listType === TEAMREFS) return TeamRefs
+  if (listType === USERS) return Users
+  if (listType === ORGANIZATIONS) return Organizations
```

Comment:
we could put these on an object and just say ` return mapper[listType]` instead of all the ifs

Reply:
yes, i was being silly there, much tidier now, i don't even need lookup, i can pass directly to list getList